uaclient/log.py

```diff
@@ -0,0 +1,57 @@
+import json
+import logging
+from collections import OrderedDict
+from typing import Any, Dict  # noqa: F401
+
+
+class JsonArrayFormatter(logging.Formatter):
+    """Json Array Formatter for our logging mechanism
+    Custom made for Pro logging needs
+    """
+
+    default_time_format = "%Y-%m-%dT%H:%M:%S%z"
+    required_fields = (
+        "asctime",
+        "name",
+        "funcName",
+        "lineno",
+        "levelname",
+        "message",
+        "extra",
+    )
+
+    def format(self, record: logging.LogRecord) -> str:
+        # format the time if is  present
+        record.message = record.getMessage()
+        if "asctime" in self.required_fields:
```

Comment:
I think we can drop this check, as we have discussed that all field should be present on an array entry